blockchain/validate.go

```diff
@@ -1817,15 +2159,16 @@ func CheckTransactionInputs(subsidyCache *SubsidyCache, tx *dcrutil.Tx, txHeight
 			return 0, ruleError(ErrDiscordantTxTree, errStr)
 		}
 
-		// The only transaction types that are allowed to spend from
-		// OP_SSTX tagged outputs are SSGen or SSRtx tx.  So, check all
-		// the inputs from non SSGen or SSRtx and make sure that they
-		// spend no OP_SSTX tagged outputs.
-		if !(isSSGen || isSSRtx) {
+		// The only transaction types that are allowed to spend from OP_SSTX
+		// tagged outputs are votes and revocations.  So, check all the inputs
+		// from non votes and revocations and make sure that they spend no
+		// OP_SSTX tagged outputs.
+		if !(isVote || isRevocation) {
 			if txscript.GetScriptClass(
```

Comment:
Do we want to eventually remove this (and the next) `GetScriptClass` and replace by more specific checks as well?

Reply:
Yeah, I think so.  I might look at that before merging this one, but it already has a lot of testing on it, so it might make sense to do that in a separate PR.